### PR TITLE
#2142: Introducing enabled attribute for CacheTagHelper.

### PR DESCRIPTION
**Pull request by sornaks**
#2142

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs [hunk 1]
```diff
@@ -108,31 +109,40 @@ public class CacheTagHelper : TagHelper
         [HtmlAttributeName(CachePriorityAttributeName)]
         public CachePreservationPriority? Priority { get; set; }
 
+        /// <summary>
+        /// Gets or sets the value that determines if the tag helper is enabled or not.
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs]: nit: "value which"

File: src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs [hunk 2]
```diff
@@ -108,31 +109,40 @@ public class CacheTagHelper : TagHelper
...
-                {
-                    UpdateCacheContext(cacheSetContext, entryLink);
-                    return result;
-                });
             }
 
             // Clear the contents of the "cache" element since we don't want to render it.
             output.SuppressOutput();
-            output.Content.SetContent(result);
+            output.Content.SetContent(result ?? await context.GetChildContentAsync());
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs]: nit: let's be more explicit about this: don't init `result`. instead, add an `else` block containing `result = await context.GetChildContentAsync();` for the disabled case.

File: test/WebSites/MvcTagHelpersWebSite/Controllers/Catalog_CacheTagHelperController.cs [hunk 3]
```diff
@@ -23,9 +23,10 @@ public ViewResult Splash(int categoryId, int correlationId, [FromHeader]string l
         }
 
         [HttpGet("/catalog/{id:int}")]
-        public ViewResult Details(int id)
+        public ViewResult Details(int id, [FromHeader]bool? isCacheEnabled)
```
Comment (dougbu) [inline test/WebSites/MvcTagHelpersWebSite/Controllers/Catalog_CacheTagHelperController.cs]: nit: space after `[FromHeader]`

File: test/WebSites/MvcTagHelpersWebSite/Views/Catalog_CacheTagHelper/Details.cshtml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 ﻿@using Microsoft.Framework.Caching.Memory
-<cache expires-after="TimeSpan.FromSeconds(1)" priority="CachePreservationPriority.Low">
+<cache expires-after="TimeSpan.FromSeconds(1)" priority="CachePreservationPriority.Low" enabled="@ViewBag.IsCacheEnabled">
```
Comment (dougbu) [inline test/WebSites/MvcTagHelpersWebSite/Views/Catalog_CacheTagHelper/Details.cshtml]: nits:
- no need for the `@`
- be consistent and use `ViewData` here too

**Comment by dougbu** [comment]
:shipit: (all nits)

File: src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs (hunk 2 above)
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs]: :+1:

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by sornaks** [comment]
Thanks! https://github.com/aspnet/Mvc/commit/b08a2154c10a3c297f1ba61138cad5726a7c7dd9

**Pull request closed without merging**